### PR TITLE
Switch gtest to re2, adjust related tests

### PR DESCRIPTION
**Pull request by jonmeow**
See https://github.com/google/googletest/blob/main/docs/advanced.md#regular-expression-syntax for gtest regex notes.

Add framework suffix handling for includes because of a dep being triggered on macos.

@micttyl FYI

### CODE REVIEW

**Comment by micttyl** [comment]
[#2181](https://github.com/carbon-language/carbon-lang/pull/2181) passes without `(.|\n)`

**Comment by micttyl** [comment]
@jonmeow  Do you mean switching or explicitly using RE2 ensures the future Windows to follow our current behavior?

**Comment by micttyl** [comment]
can you try `(.|\n)` -> `.+\n`

**Comment by micttyl** [comment]
This is a superior alternative to [2181](https://github.com/carbon-language/carbon-lang/pull/2181)

**Comment by jonmeow** [comment]
> @jonmeow Do you mean switching or explicitly using RE2 ensures the future Windows to follow our current behavior? Well, that could be after LLVM supporting Windows.

Yes, RE2 has cross-platform compatibility.

Note, LLVM supports Windows, it's only Carbon's BUILD that has trouble.

> [#2181](https://github.com/carbon-language/carbon-lang/pull/2181) passes without `(.|\n)`
>
> can you try `(.|\n)` -> `.+\n`

re2 defaults to `.` not matching newlines (s = false), and there are multiple lines being matched here. re2 syntax documentation is at https://github.com/google/re2/wiki/Syntax

File: bazel/cc_toolchains/clang_configuration.bzl [hunk 1]
```diff
@@ -112,8 +112,10 @@ def _compute_clang_cpp_include_search_paths(repository_ctx, clang, sysroot):
     include_begin = output.index("#include <...> search starts here:") + 1
     include_end = output.index("End of search list.", include_begin)
 
+    # Suffix present on framework paths.
+    framework_suffix = " (framework directory)"
     return [
-        repository_ctx.path(s.lstrip(" "))
+        repository_ctx.path(s.lstrip(" ").removesuffix(framework_suffix))
```
Comment (chandlerc) [inline bazel/cc_toolchains/clang_configuration.bzl:118]: I think this is an unrelated change that should be landed in its own PR?

**Review by chandlerc — APPROVED** [review]
LG to submit without the framework suffix fix, as I thought that was unrelated.

If its actually related to the googletest update, my bad, feel free to submit as-is.

File: bazel/cc_toolchains/clang_configuration.bzl (hunk 1 above)
Comment (jonmeow) [reply bazel/cc_toolchains/clang_configuration.bzl:118]: To try to explain this -- I think the googletest update tickles a new dependency on absl that depends on the framework code. So this is necessary for this change.

Comment (jonmeow) [reply bazel/cc_toolchains/clang_configuration.bzl:118]: Added a note to the PR description.
